### PR TITLE
Adds method to force regeneration of certs

### PR DESCRIPTION
**Pull request by chuckbutler**
WIP - For talking points presently 

Introduces the 'tls.regenerate_certificates' state, which allows consumer layers to reconfigure the primary ssl configuration certificate (on the master)

This has a side-effect of removing the flat file PKI database that is initially generated during TLS setup. This is unfortunate, but is a work-around to conform to expectations of the leader.

Follower behavior should be unchanged with this behavior.

### CODE REVIEW

**Comment by chuckbutler** [comment]
This also seems to correct the behavior of #22 when adding the force param.

File: reactive/tls.py [hunk 1]
```diff
@@ -21,6 +21,19 @@
 from contextlib import contextmanager
 
 
+@when('tls.regenerate_certificates')
+def regenerate_certificates():
+    ''' Allow the calling layer to modify settings, and trigger the
+        certificates to be re-generated (perhaps we updated the openssl conf)
+    '''
+    print(' ==> regenerating certificates')
```
Comment (mbruzek) [inline reactive/tls.py]: replace this print with a log message.

**Comment by mbruzek** [comment]
I took a preliminary look at this code change. It seems a bit drastic to delete the entire easy-rsa directory to configure easy-rsa. Perhaps the reason you were having trouble configuring easy-rsa before it generated the certificate is the reactive framework runs all valid states asynchronously, the theory being your valid state was racing with the ca certificate creation.

As an alternate solution: How about adding an intermediary state between install() and the check_ca_status() method. This state could be triggered when `easy-rsa installed` and do some configuration. And then change the check_ca_status() to use the decorator @when('easy-rsa configured') or something like that? If that is not possible I would be curious to why it does not work.

**Comment by chuckbutler** [comment]
@mbruzek  - Point taken regarding wholesale removal of the PKI DB. After some additional poking, this has been refactored to introduce and use an intermediary state which will invoke the proper reactive handler in the calling layer (in this case, swarm) letting it reconfigure the easyrsa configuration prior to generating the certs, without deleting anything.

can you TAL?

File: reactive/tls.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 
 from charmhelpers.core import hookenv
 from charmhelpers.core import unitdata
+from charmhelpers.core.hookenv import log
```
Comment (mbruzek) [inline reactive/tls.py]: Minor nit, I don't see any of your new code using the log import. pep8 would not approve

**Comment by mbruzek** [comment]
One minor nit that should not prevent this code from going in. I did not test this code, but watched it work on @chuckbutler 's console.

**Pull request merged**